framework/src/wifi_manager/wifi_manager_api.c

```diff
@@ -135,79 +152,73 @@ wifi_manager_result_e wifi_manager_set_mode(wifi_manager_mode_e mode, wifi_manag
 
 	sem_t signal;
 	sem_init(&signal, 0, 0);
-	wifimgr_msg_s msg = {EVT_SET_STA_CMD, NULL, &signal};
+	wifimgr_msg_s msg = {EVT_SET_STA_CMD, WIFI_MANAGER_FAIL, NULL, &signal};
 	if (mode == SOFTAP_MODE) {
 		msg.event = EVT_SET_SOFTAP_CMD;
 		msg.param = (void *)config;
 	}
 
-	wifi_manager_result_e res = wifimgr_post_message(&msg);
+	int res = wifimgr_post_message(&msg);
```

Comment:
Here the same issue mentioned above.
If 'res' is less than zero, do you need to still wait 'msg.signal'?